### PR TITLE
fix(Controls) avoid sliding with stroke uniform

### LINKED ISSUE #9423: [Bug]: The polyline origin position keeps moving when I move the point after scale if set strokeUniform = true
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.0-beta13

### In What environments are you experiencing the problem?

Microsoft Edge

### Node Version (if applicable)

None

### Link To Reproduction

https://codepen.io/ritali4912/pen/WNLqEXe

### Steps To Reproduce

1. Create a polyline, set the strokeUniform = true
2. Scale the polyline. (Caused there is a reference line, I scale the tr control)
3. Then keep dragging any point control.
4. The polyline position changing.

### Expected Behavior

The polyline position keeps no change.

### Actual Behavior

**At beginning**
image

**Scale the tr control**
image

**dragging the 3rd point control**
image

### Error Message & Stack Trace

_No response_

### PR DESCRIPTION
**Pull request by asturur**
## Motivation

As reported in #9423 , there is an issue in determining the correct anchor point when strokeUniform is used.
This PR propose a fix by excluding strokeWidth by the calculations when determining the anchor point.

closes #9423

### CODE REVIEW

**Comment by asturur** [comment]
MIssing code cleanup and interaction test.

**Comment by asturur** [comment]
also polygons with skew do not work ok.
But that could be explictly not supported

**Comment by ShaMan123** [comment]
You blocked that fix

**Comment by ShaMan123** [comment]
I have given a quick read
This seems a patch, not a fix
The real issue is that stroke is considered for layout calculations.
That is completely wrong. And has many related issues.
Stroke should be considered for bbox calculation only.

**Comment by asturur** [comment]
> You blocked that fix

it wasn't linked in the issue or mentioned, can you point me to it?

**Comment by asturur** [comment]
> I have given a quick read This seems a patch, not a fix The real issue is that stroke is considered for layout calculations. That is completely wrong. And has many related issues. Stroke should be considered for bbox calculation only.

This the wrong approach imho.
There is a function here that has to calculate an anchor point, it is isolated, and is not the base for other functionalities.
The function doesn't work and a fix that is scoped to the function only is a good fix.
As you say if stroke ( with strokeUniform ) shouldn't considered for real calculation, removing stroke from the calculation when necessary is a solution.

strokeUniform is a weird functionality to support,  the real fix would be removing strokeUniform, so stroke can stay happily in calculation.

**Comment by ShaMan123** [comment]
This is a discussion to conduct thoroughly.

**Comment by ShaMan123** [comment]
It is a major part of the design of fabric and the coords/layout system

**Comment by ShaMan123** [comment]
strokeUniform will not be that much of a pain if stroke is not part of layout

**Comment by ShaMan123** [comment]
> > You blocked that fix
> 
> it wasn't linked in the issue or mentioned, can you point me to it?

https://github.com/fabricjs/fabric.js/pull/8556#pullrequestreview-1234400096
https://github.com/fabricjs/fabric.js/pull/8556#issuecomment-1373177678
https://github.com/fabricjs/fabric.js/pull/8556#issuecomment-1375175788

**Comment by asturur** [comment]
This needs to be re-examined after #9460 is merged.
I can't mix and match the fixes now for time constrains, but if after #9460 skewX can work we need to update one test to take care of it.

**Comment by asturur** [comment]
> > > You blocked that fix
> > 
> > 
> > it wasn't linked in the issue or mentioned, can you point me to it?
> 
> [#8556 (review)](https://github.com/fabricjs/fabric.js/pull/8556#pullrequestreview-1234400096) [#8556 (comment)](https://github.com/fabricjs/fabric.js/pull/8556#issuecomment-1373177678) [#8556 (comment)](https://github.com/fabricjs/fabric.js/pull/8556#issuecomment-1375175788)
> 
> [e6ba30c](https://github.com/fabricjs/fabric.js/commit/e6ba30c53506e871de77bd848144b4711d8dc21b)

oh now i understand what you mean.
I thought you were talking about this fix blocked somewhere else.

Yes i still think skewing is an edge case and if you want to support it with polygon editing you are better off merging skew in the points every time you apply it.
The effect is the same and everything is so much easier.

File: src/controls/polyControl.ts
Comment (ritali4912) [inline src/controls/polyControl.ts:1]: If a two points horizontal polyline with strokeWidth = 0, it's height will be 0. Does the divide in below got NaN?
const newPositionNormalized = anchorPoint
      .subtract(poly.pathOffset)
      **.divide(poly._getNonTransformedDimensions())**
      .multiply(adjustFlip);

File: src/controls/polyControl.ts [hunk 1]
```diff
@@ -80,7 +80,12 @@ export const factoryPolyActionHandler = (
     y: number
   ) {
     const poly = transform.target as Polyline,
-      anchorPoint = new Point(
+      strokeWidth = poly.strokeWidth;
+    // do all calculation without strokeWidth
+    // to avoid scaling issues between dimensions and scaling factors
+    // when strokeUniform is set as true
+    poly.strokeWidth = 0;
```
Comment (ShaMan123) [inline src/controls/polyControl.ts:87]:
```ts
getStrokeWidthVector(strokeWidth = this.strokeWidth) {
    const strokeWidthFactor = new fabric.Point(strokeWidth, strokeWidth);
    return this.strokeUniform
      ? strokeWidthFactor.divide(this.getObjectScaling())
      : strokeWidthFactor;
  }
```

**Comment by asturur** [comment]
this is obsolete

**Pull request closed without merging**